modules/core/shared/src/main/scala/io/circe/Codec.scala

```diff
@@ -26,7 +26,13 @@ trait Codec[A] extends Decoder[A] with Encoder[A] {
 }
 
 object Codec extends ProductCodecs with EnumerationCodecs {
-  def apply[A](implicit instance: Codec[A]): Codec[A] = instance
+  def apply[A](implicit decoder: Decoder[A], encoder: Encoder[A]): Codec[A] =
+    decoder match {
+      case codec: Codec[_] if codec eq encoder =>
+        // if codec: Decoder[A] with Decoder[B], A and B must be identical
```

Comment:
is this comment correct? 
Should not this be something like:
// if codec: Decoder[A] with Encoder[A], encoder and decoder must be the same instance

Reply:
Same situation: https://www.scala-lang.org/old/node/3423

Reply:
@hamnis

Reply:
Well yes, this is expected.

Why one would extend the same type twice is basically what my comment is about.